### PR TITLE
Filter implemented interfaces through the type blacklist.

### LINKED ISSUE #421: @implements ignores the type blacklist
Because it doesn't use `type-translator.ts`. That's a bit of a code smell, but probably still better than trying to use it and ending up with something like `@implements {?}`. But maybe we should still centralize the code doing type emits into `type-translator.ts`.

### PR DESCRIPTION
**Pull request by mprobst**
Fixes #421.

### CODE REVIEW

File: test_files/jsdoc_types/jsdoc_types.ts [hunk 1]
```diff
@@ -31,3 +31,7 @@ let useDefaultClassAsType: DefaultClass;
 // NeverTyped should be {?}, even in typed mode.
 let useNeverTyped: NeverTyped;
 let useNeverTyped2: string|NeverTyped;
+/** Note: no implements JSDoc clause because the type is blacklisted. */
+class ExtendsNeverTyped implements NeverTyped {
```
Comment (evmar) [inline test_files/jsdoc_types/jsdoc_types.ts]: I guess ImplementsNeverTyped, not Extends.

File: src/tsickle.ts [hunk 2]
```diff
@@ -920,6 +925,9 @@ class Annotator extends ClosureRewriter {
             if (sym.flags & ts.SymbolFlags.Alias) {
               sym = typeChecker.getAliasedSymbol(sym);
             }
+            if (this.newTypeTranslator(heritage).isBlackListed(sym)) {
```
Comment (evmar) [inline src/tsickle.ts]: Is it important that this use a different context parameter than all the other uses of typetranslator?

Comment (mprobst) [inline src/tsickle.ts]: Nope, I slipped in finding the closest context node (not that it matters). Fixed.

File: test_files/jsdoc_types/jsdoc_types.ts (hunk 1 above)
Comment (mprobst) [inline test_files/jsdoc_types/jsdoc_types.ts]: Done.

**Pull request merged**